src/main/java/camp/nextstep/edu/rebellion/calc/Calculator.java

```diff
@@ -0,0 +1,53 @@
+package camp.nextstep.edu.rebellion.calc;
+
+import camp.nextstep.edu.rebellion.calc.util.NumberUtil;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Calculator {
+    private static final String DELIMITER = " ";
+
+    private Calculator() {}
+
+    public static int calc(String expression) {
+        List<String> expressions = Arrays.asList(expression.split(DELIMITER));
+        List<Integer> numbers = parseNumbers(expressions);
+        List<CalculationOperator> operators = parseOperators(expressions);
+
+        validate(numbers, operators);
+
+        return makeResult(numbers, operators);
+    }
+
+    private static List<Integer> parseNumbers(List<String> expressions) {
+        return expressions.stream()
+                .filter(NumberUtil::isNumeric)
+                .map(Integer::parseInt)
+                .collect(Collectors.toList());
+    }
+
+    private static List<CalculationOperator> parseOperators(List<String> expressions) {
+        return expressions.stream()
+                .filter(s -> !NumberUtil.isNumeric(s))
+                .map(s -> CalculationOperator.of(s))
+                .collect(Collectors.toList());
+    }
+
+    public static void validate(List<Integer> numbers, List<CalculationOperator> operators) {
+        if(numbers.size() == operators.size() + 1) {
+            return;
+        }
+        throw new IllegalArgumentException("수식이 올바르지 않습니다");
+    }
+
+    private static int makeResult(List<Integer> numbers, List<CalculationOperator> operators) {
+        int idx = 0;
+        int result = numbers.get(0);
+        for(int i = 1; i < numbers.size(); i++) {
+            result = operators.get(idx++).operation(result, numbers.get(i));
+        }
+        return result;
+    }
```

Comment:
숫자와 연산자를 구별하여 관리해주셨고 람다식으로 구현하셨기에 해당 로직도 
다음과 같은 람다식으로도 변경해볼 수 있을 것 같아요 :)
참고만 해주시면 감사하겠습니다 😄 
```suggestion
    private static int makeResult(List<Integer> numbers, List<CalculationOperator> operators) {
        int[] index = { 0 };
        int result = numbers.stream()
                            .reduce((preOperand, postOperand) -> {
                                 int sum = operators.get(index[0]).operation(preOperand, postOperand);
                                 index[0] += 1;
                                 return sum;
                            }).get();
        return result;
    }
```

Reply:
진짜 감사합니다 😭 제가 저걸 해보고 싶어서 오늘 아침부터 계속 삽질했는데 
명쾌하게 알려주셔서 감사합니다 👍